dynamiq/nodes/images/edit.py

```diff
@@ -0,0 +1,311 @@
+
+    if original_format in ("JPEG", "JPG"):
+        if img_obj.mode not in ("RGB", "L"):
+            img_obj = img_obj.convert("RGB")
+    else:
+        if img_obj.mode not in ("RGBA", "LA", "L"):
+            img_obj = img_obj.convert("RGBA")
+
+    output_bytes = io.BytesIO()
+    img_obj.save(output_bytes, format=original_format)
+    output_bytes.seek(0)
+    return output_bytes
+
+
+class ImageEditInputSchema(BaseModel):
+    """Input schema for image editing."""
+
+    prompt: str = Field(..., description="Text prompt describing the desired edits.")
+    files: list[io.BytesIO | bytes] | io.BytesIO | bytes = Field(
+        ...,
+        description="The image(s) to edit. Can be a single image or a list of images. Auto-injected from agent's "
+        "file store.",
+        json_schema_extra={"map_from_storage": True, "is_accessible_to_agent": False},
+    )
+    mask: io.BytesIO | bytes | None = Field(
+        default=None,
+        description="Optional mask image indicating areas to edit.",
+    )
+    n: int | None = None
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+class ImageEdit(ConnectionNode):
+    """
+    Node for editing images using AI models.
+
+    Takes an existing image and a text prompt to generate edited versions.
+    Optionally accepts a mask to specify which areas to modify.
+
+    Attributes:
+        name (str): The name of the node.
+        model (str): The model to use for image editing (e.g., 'dall-e-2', 'gpt-image-1').
+        connection (OpenAIConnection): The connection to the API.
+        n (int): Number of edited images to generate.
+        size (ImageSize | str): Size of the output images.
+        response_format (ImageResponseFormat | str | None): Response format (e.g., 'url', 'b64_json'). Only supported
+        by some models.
+    """
+
+    group: Literal[NodeGroup.IMAGES] = NodeGroup.IMAGES
+    name: str = "Image Edit"
+    description: str = """Edit and modify existing images using AI with text prompts and optional masking.
+
+Key Capabilities:
+- AI-powered image editing with natural language prompts
+- Selective area editing using optional mask images
+- Multiple variations generation (set n parameter)
+- Configurable output sizes (256x256 to 1792x1024)
+- URL or base64 JSON response formats
+- Preserves original filenames in outputs
+
+Usage Strategy:
+- Provide clear, descriptive prompts for desired edits
+- Use masks to target specific areas for modification
+- Generate multiple variations to explore different results
+- Integrate seamlessly with agent workflows via file storage
+
+Parameter Guide:
+- prompt: Text description of desired edits (required)
+- files: Image file/files to edit, auto-injected from agent's file store
+- mask: Optional mask image to specify areas to edit
+- n: Number of edited versions to generate
+- size: Output dimensions (e.g., '1024x1024', '1792x1024')
+- response_format: 'url' or 'b64_json' output format
```

Comment:
- `using AI` at line 107 is redundant
- `Preserves original filenames in outputs` is not necessary (line 115)
- `Integrate seamlessly with agent workflows via file storage` (line 121) is not needed